app/src/main/java/com/epfl/drawyourpath/path/RunsAdapter.kt

```diff
@@ -45,12 +46,10 @@ class RunsAdapter(private var runs: List<Run>) : RecyclerView.Adapter<RunsAdapte
         // Set the data to the view items in the layout
         // holder.mapImageView.setImageResource(run.mapImage)
         holder.dateTextView.text = run.getDate()
-        holder.distanceTextView.text =
-            "Distance: ${String.format("%.2f", run.getDistance() / 1000)} Km"
-        holder.timeTakenTextView.text = "Time taken: ${run.getDuration() / 60} minutes"
-        holder.calorieTextView.text = "Calories burned: ${run.getCalories()} kcal"
-        holder.averageSpeedTextView.text =
-            "Speed: ${String.format("%.2f", run.getAverageSpeed())} m/s"
+        holder.distanceTextView.text = holder.context.getString(R.string.display_distance).format(run.getDistance() / 1000)

```

Comment:
You have an helper function in util to directly display the distance in a correct format.